src/__tests__/lib/containers/table/datasets/DatasetItemsEditor.test.tsx

```diff
@@ -0,0 +1,564 @@
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils'
+import {
+  EntityFinderModal,
+  EntityFinderModalProps,
+} from '../../../../../lib/containers/entity_finder/EntityFinderModal'
+import {
+  DatasetItemsEditor,
+  DatasetItemsEditorProps,
+} from '../../../../../lib/containers/table/datasets/DatasetItemsEditor'
+import { displayToast } from '../../../../../lib/containers/ToastMessage'
+import { createWrapper } from '../../../../../lib/testutils/TestingLibraryUtils'
+import { ENTITY_ID } from '../../../../../lib/utils/APIConstants'
+import {
+  BackendDestinationEnum,
+  getEndpoint,
+} from '../../../../../lib/utils/functions/getEndpoint'
+import { SynapseContextType } from '../../../../../lib/utils/SynapseContext'
+import { DatasetItem, Reference } from '../../../../../lib/utils/synapseTypes'
+import {
+  mockDatasetEntity,
+  mockFileEntity,
+  MOCK_DATASET_ENTITY_ID,
+} from '../../../../../mocks/entity/mockEntity'
+import { rest, server } from '../../../../../mocks/msw/server'
+
+jest.mock('../../../../../lib/containers/ToastMessage', () => {
+  return { displayToast: jest.fn() }
+})
+
+const mockFileReference: Reference = {
+  targetId: mockFileEntity.id!,
+  targetVersionNumber: 3,
+}
+
+function referenceToDatasetItem(reference: Reference): DatasetItem {
+  return {
+    entityId: reference.targetId,
+    versionNumber: reference.targetVersionNumber!,
+  }
+}
+// The Entity Finder is complicated to use and would require setting up a lot of API mocks, so we'll just mock the component.
+jest.mock(
+  '../../../../../lib/containers/entity_finder/EntityFinderModal',
+  () => {
+    return {
+      EntityFinderModal: jest.fn(() => <></>),
+    }
+  },
+)
+const mockEntityFinder = EntityFinderModal as unknown as jest.Mock<
+  (props: Partial<EntityFinderModalProps>) => JSX.Element
+>
+const mockEntityFinderButtonText = 'Add Items From Entity Finder'
+function mockEntityFinderToAddItems(items: Array<Reference>) {
+  mockEntityFinder.mockImplementation(({ show, onConfirm }) => {
+    return (
+      <>
+        {show ? (
+          <div role="dialog">
+            <button onClick={() => onConfirm(items)}>
+              {mockEntityFinderButtonText}
+            </button>
+          </div>
+        ) : null}
+      </>
+    )
+  })
+}
+
+function addItemsViaEntityFinder() {
```

Comment:
It seems like this function is just testing your mock entity finder implementation (and the ability to find the Add Items button on the screen).

Reply:
Wait, is this essentially just a utility function to invoke the onConfirm() function given to the entity finder modal

Reply:
It is basically a utility function. I could add a comment or remove the `expect`s to make that more clear